openshift/commonPipeline.groovy

```diff
@@ -0,0 +1,253 @@
+            throw e
+          }
+        }
+      },
+
+      Frontend: {
+        dir('app/frontend') {
+          try {
+            echo 'Installing NPM Dependencies...'
+            sh 'npm ci'
+
+            echo 'Linting and Testing Frontend...'
+            sh 'npm run test'
+
+            echo 'Frontend Lint Checks and Tests passed'
+          } catch (e) {
+            echo 'Frontend Lint Checks and Tests failed'
+            throw e
+          }
+        }
+      }
+    )
+  }
+}
+
+// Build Images & SonarQube Report
+def runStageBuild() {
+  openshift.withCluster() {
+    openshift.withProject(TOOLS_PROJECT) {
+      if(DEBUG_OUTPUT.equalsIgnoreCase('true')) {
+        echo "DEBUG - Using project: ${openshift.project()}"
+      }
+
+      parallel(
+        App: {
+          try {
+            notifyStageStatus('App', 'PENDING')
+
+            echo "Processing BuildConfig ${REPO_NAME}-app-${JOB_NAME}..."
+            def bcApp = openshift.process('-f',
+              'openshift/app.bc.yaml',
+              "REPO_NAME=${REPO_NAME}",
+              "JOB_NAME=${JOB_NAME}",
+              "SOURCE_REPO_URL=${SOURCE_REPO_URL}",
+              "SOURCE_REPO_REF=${SOURCE_REPO_REF}"
+            )
+
+            echo "Building ImageStream..."
+            openshift.apply(bcApp).narrow('bc').startBuild('-w').logs('-f')
+
+            echo "Tagging Image ${REPO_NAME}-app:latest..."
+            openshift.tag("${REPO_NAME}-app:latest",
+              "${REPO_NAME}-app:${JOB_NAME}"
+            )
+
+            echo 'App build successful'
+            notifyStageStatus('App', 'SUCCESS')
+          } catch (e) {
+            echo 'App build failed'
+            notifyStageStatus('App', 'FAILURE')
+            throw e
+          }
+        },
+
+        SonarQube: {
+          unstash APP_COV_STASH
+          unstash FE_COV_STASH
+
+          echo 'Performing SonarQube static code analysis...'
+          sh """
+          sonar-scanner \
+            -Dsonar.host.url='${SONARQUBE_URL_INT}' \
+            -Dsonar.projectKey='${REPO_NAME}-${JOB_NAME}' \
+            -Dsonar.projectName='${APP_NAME} (${JOB_NAME.toUpperCase()})'
+          """
+        }
+      )
+    }
+  }
+}
+
+// Deploy Application and Dependencies
+def runStageDeploy(String stageEnv, String projectEnv, String hostEnv, String pathEnv) {
+  if (!stageEnv.equalsIgnoreCase('Dev')) {
+    input("Deploy to ${projectEnv}?")
+  }
+
+  openshift.withCluster() {
+    openshift.withProject(projectEnv) {
+      if(DEBUG_OUTPUT.equalsIgnoreCase('true')) {
+        echo "DEBUG - Using project: ${openshift.project()}"
+      }
+
+      notifyStageStatus("Deploy - ${stageEnv}", 'PENDING')
+      createDeploymentStatus(projectEnv, 'PENDING', JOB_NAME, hostEnv, pathEnv)
+
+      echo "Checking for ConfigMaps and Secrets in project ${openshift.project()}..."
+      if(!(openshift.selector('cm', "getok-frontend-config").exists() &&
+      openshift.selector('cm', "getok-oidc-config").exists() &&
```

Comment:
Is this one still used in the new model?

Reply:
It is not. However, until we merge to master, I want to leave these checks in place as deleting these config maps will make the master deploy fail to run. At this point, it's more for deployment safety insurance than anything else.